scripts/run_forge.sh

```
@@ -0,0 +1,164 @@
+fi
+FORGE_NAMESPACE=${FORGE_NAMESPACE:-$namespace}
+# clean up namespace name
+# replace non alphanumeric chars with dash
+FORGE_NAMESPACE="${FORGE_NAMESPACE//[^[:alnum:]]/-}"
+# use the first 64 chars only for namespace, as it is the maximum for k8s resources
+FORGE_NAMESPACE=${FORGE_NAMESPACE:0:64}
+
+
+if [ "$FORGE_NAMESPACE_KEEP" = "true" ]; then
+    KEEP_ARGS="--keep"
+fi
+
+if [ "$FORGE_ENABLE_HAPROXY" = "true" ]; then
+    ENABLE_HAPROXY_ARGS="--enable-haproxy"
+fi
+
+
+if [ -z "$IMAGE_TAG" ]; then
+    echo "IMAGE_TAG not set"
+    exit 1
+fi
+
+echo "Ensure image exists"
+img=$(aws ecr describe-images --repository-name="aptos/validator" --image-ids=imageTag=$IMAGE_TAG)
+if [ $? != 0 ]; then
+    echo "IMAGE_TAG does not exist: ${IMAGE_TAG}"
+    exit 1
+fi
+
+FORGE_CLUSTER_NAME=$(kubectl config current-context | grep -oE 'aptos.*')
+
+echo "Using cluster ${FORGE_CLUSTER_NAME} from current kubectl context"
+
+FORGE_START_TIME_MS="$(date '+%s')000"
+
+# # Run forge with test runner in k8s
+if [ "$FORGE_RUNNER_MODE" = "local" ]; then
+
+    # more file descriptors for heavy txn generation
+    ulimit -n unlimited
+
+    cargo run -p forge-cli -- test k8s-swarm \
+        --image-tag $IMAGE_TAG \
+        --namespace $FORGE_NAMESPACE \
+        --port-forward $KEEP_ARGS $ENABLE_HAPROXY_ARGS | tee $FORGE_OUTPUT
+
+    FORGE_EXIT_CODE=$?
+
+    # try to kill orphaned port-forwards
+    if [ -z "$KEEP_ARGS" ]; then
+        ps -A | grep  "kubectl port-forward -n $FORGE_NAMESPACE" | awk '{ print $1 }' | xargs -I{} kill -9 {}
+    fi
+
+else
+    # try deleting existing forge pod of same name
+    # since forge test runner will run in a pod that matches the namespace
+    # this will pre-empt the existing forge test in the same namespace and ensures
+    # we do not have any dangling test runners
+    FORGE_POD_NAME=$FORGE_NAMESPACE
+    kubectl delete pod $FORGE_POD_NAME || true
+    kubectl wait --for=delete "pod/${FORGE_POD_NAME}" || true
```

why || true (i.e. ignoring the error) ?

---

Those few lines delete existing Forge test runner pods of the same name, which is helpful when CI runs Forge a bunch of times in a row. However, there's no guarantee that these pods actually exist (pods have either been cleaned up or it is indeed a fresh run), so those steps are allowed to fail without blocking starting the actual test